pkg/sources/gitlab/gitlab.go

```diff
@@ -81,6 +84,57 @@ func (s *Source) JobID() sources.JobID {
 	return s.jobID
 }
 
+// filteredRepoCache is a wrapper around cache.Cache that filters out repos
+// based on include and exclude globs.
+type filteredRepoCache struct {
+	cache.Cache[string]
```

Comment:
It doesn't look like the cache part of `filteredRepoCache` is ever used, so I don't think this is really a cache! That's a good thing - the repo caching in the GitHub source is pretty complicated, and it's good that we don't have to do it here too. You can remove the embedded `cache.Cache` and rename this structure - or even eliminate it entirely and pass the include/exclude globs directly to `buildIgnorer` (which would mean making `buildIgnorer` a little more complicated, but would also mean that you don't have to make anything _else_ more complicated).

Reply:
@rosecodym the idea is to compile it very first time and just reuse it, similar to Github. Moving compilation in buildignorer will compile globs for all kind of Projects (user projects, group projects etc).

Reply:
btw I do agree that `cache.Cache` is not needed in Gitlab source. I will take a look on that.

Reply:
@rosecodym  I have removed the concept of caching and renamed the entity 'globRepoFilter'. I did not eliminate the whole structure as I believe having that is just making the code more clear. Let me know what are your thoughts on this.